File: src/js/gen-img.js
const { OpenAI } = require('openai');

const openai = new OpenAI('your-api-key'); // Replace 'your-api-key' with your actual API key

async function generateImage() {
  try {
    const response = await openai.createImage({
      model: "dall-e-3",
      prompt: "a white siamese cat",
      n: 1,
      quality: "hd",
      size: "1024x1024",
    });
    const image_url = response.data.data[0].url;
    console.log('Generated image URL:', image_url);
    return image_url;
  } catch (error) {
    console.error('Error generating image:', error);
    throw error; // Re-throw the error to handle it outside this function if needed
  }
}

// Call the asynchronous function
generateImage()
  .then(image_url => {
    // Handle the generated image URL
    console.log('Generated image URL:', image_url);
  })
  .catch(error => {
    // Handle errors
    console.error('Error:', error);
  });

File: src/py/genimg.py
from openai import OpenAI

client = OpenAI()

try:
    # Make API request
    response = client.images.generate(
        model="dall-e-3",
        prompt="Tiny hand in mine,Strength grows with each wobbly step, My heart, forever his",
        size="1024x1024",
        quality="hd",
        n=1,
    )

    # Check if response object is valid
    if response is not None:
        # Check if response contains data
        if len(response.data) > 0:
            # Check if data contains image URL
            if hasattr(response.data[0], 'url'):
                # Get and print image URL
                image_url = response.data[0].url
                print("Generated Image URL:", image_url)
            else:
                print("No image URL found in the response.")
        else:
            print("No data returned in the response.")
    else:
        print("No response received from the API.")

except Exception as e:
    # Print any errors that occur
    print("An error occurred:", str(e))